packages/contracts/test/EmailAuth.t.sol

```diff
@@ -0,0 +1,133 @@
+        emailAuth.updateDKIMRegistry(address(dkim));
+
+        
+        templateId = 1;
+        subjectTemplate = ["Send", "{decimals}", "ETH", "to", "{string}"];
+        newSubjectTemplate = ["Send", "{decimals}", "USDC", "to", "{string}"];
+    }
+
+    function testDkimRegistryAddr() public view {
+        address dkimAddr = emailAuth.dkimRegistryAddr();
+        assertEq(dkimAddr, address(dkim));
+    }
+    
+    function testVerifierAddr() public view {
+        address verifierAddr = emailAuth.verifierAddr();
+        assertEq(verifierAddr, address(verifier));
+    }
+
+    function testUpdateDKIMRegistry() public {
+        ECDSAOwnedDKIMRegistry newDKIM = new ECDSAOwnedDKIMRegistry(msg.sender);
+        emailAuth.updateDKIMRegistry(address(newDKIM));
+        assertEq(emailAuth.dkimRegistryAddr(), address(newDKIM));
+    }
+    
+    function testUpdateVerifier() public {
+        Verifier newVerifier = new Verifier();
+        emailAuth.updateVerifier(address(newVerifier));
+        assertEq(emailAuth.verifierAddr(), address(newVerifier));
+    }
+
+    function testInsertSubjectTemplate() public {        
+        emailAuth.insertSubjectTemplate(templateId, subjectTemplate);
+    }
+
+    function testUpdateSubjectTemplate() public {
+        this.testInsertSubjectTemplate();
+        emailAuth.updateSubjectTemplate(templateId, newSubjectTemplate);
+    }
+
+    function testDeleteSubjectTemplate() public {
+        this.testInsertSubjectTemplate();
+        emailAuth.deleteSubjectTemplate(templateId);
+    }
+
+    function testComputeMsgHash() public view {
+        bytes[] memory subjectParams = new bytes[](2);
+        subjectParams[0] = abi.encode(1);
+        subjectParams[1] = abi.encode(vm.addr(1));
+        bytes32 msgHash = emailAuth.computeMsgHash(accountSalt, true, templateId, subjectParams);
+        assertEq(msgHash, 0xe7f92de45c5ad2716ecefebf444d57aba480764d104b92d2a1bb7820f344bb22);
+    }
+
+    function testAuthEmail() public {
+        this.testInsertSubjectTemplate();
+
+        bytes[] memory subjectParams = new bytes[](2);
+        subjectParams[0] = abi.encode(1);
+        subjectParams[1] = abi.encode("alice@gmail.com");
+
+        EmailProof memory emailProof = EmailProof({
+            domainName: "gmail.com",
+            publicKeyHash: 0x0ea9c777dc7110e5a9e89b13f0cfc540e3845ba120b2b6dc24024d61488d4788,
+            timestamp: 1694989812,
+            maskedSubject: "Send 1 ETH to alice@gmail.com",
+            emailNullifier: 0x00a83fce3d4b1c9ef0f600644c1ecc6c8115b57b1596e0e3295e2c5105fbfd8a,
+            accountSalt: accountSalt,
+            isCodeExist: true,
+            proof: mockProof
+        });
+
+        EmailAuthMsg memory emailAuthMsg = EmailAuthMsg({
+            templateId: templateId,
+            subjectParams: subjectParams,
+            skipedSubjectPrefix: 0,
+            proof: emailProof
+        });
+
+        // TODO: This mockCall should be removed
```

Comment:
Because of the use of mockProof, verifyEmailProof will always fail. That's the reason I'm using mockCall to ensure that this call always succeeds.
I would like to remove the mockCall if I have the correct proof.
Is that possible?

Reply:
For the unit tests, I think it is fine to use a mock verifier.
I will prepare the proving keys for the integration tests.